temporalio/workflow.py

```diff
@@ -5074,3 +5152,71 @@ def _to_proto(self) -> temporalio.bridge.proto.common.VersioningIntent.ValueType
         elif self == VersioningIntent.DEFAULT:
             return temporalio.bridge.proto.common.VersioningIntent.DEFAULT
         return temporalio.bridge.proto.common.VersioningIntent.UNSPECIFIED
+
+
+# Nexus
+
+ServiceT = TypeVar("ServiceT")
+
+
+class NexusClient(Generic[ServiceT]):
+    def __init__(
+        self,
+        service: Union[Type[ServiceT], str],
+        *,
+        endpoint: str,
```

Comment:
It is a bit confusing when creating the Nexus client to provide the endpoint _after_ the service (usually you go least specific to most specific and other SDKs do this as well). I understand there may be a task queue instead of an endpoint one day. 

I also think we don't want users instantiating the client but instead it should be abstract and an internal impl that is only obtainable via a call on the workflow module.

So how about making this an ABC and having:

```python
def create_nexus_client(endpoint: str, service: Union[Type[ServiceT], str]) -> NexusClient[ServiceT]
```

Which matches Go, and if we have a task queue form we can provide an overload or something?

Reply:
I'm not sure that endpoint should come before service. They are creating the Nexus client _for_ a specific service. endpoint feels more like metadata.

Also using overloads for alternative positional signatures like that isn't Pythonic and relies on types. Python users will expect optional kwargs rather than multiple mutually inconsistent overloads.

Are you sure it shouldn't be `create_nexus_client(service, endpoint=my_endpoint)` and then `create_nexus_client(service, task_queue=my_task_queue)`?

Reply:
I've made them both kwarg-only pending agreement on positionals.